src/sources/core_text.rs

```diff
@@ -146,9 +147,45 @@ fn create_handles_from_core_text_collection(
 ) -> Result<Vec<Handle>, SelectionError> {
     let mut fonts = vec![];
     if let Some(descriptors) = collection.get_descriptors() {
-        for index in 0..descriptors.len() {
+        let mut font_data = HashMap::new();
+
+        'outer: for index in 0..descriptors.len() {
             let descriptor = descriptors.get(index).unwrap();
-            fonts.push(create_handle_from_descriptor(&*descriptor));
+            let postscript_name = descriptor.font_name();
+            let font_path = descriptor.font_path().unwrap();
+
+            let data = if let Some(data) = font_data.get(&font_path) {
+                Arc::clone(data)
+            } else {
+                let mut file = if let Ok(file) = File::open(&font_path) {
+                    file
+                } else {
+                    break;
+                };
+                let data = if let Ok(data) = utils::slurp_file(&mut file) {
+                    Arc::new(data)
+                } else {
+                    break;
```

Comment:
Do we actually want to break here, rather than continuing?

Reply:
Thanks for your review. Yeah, we should use continuing here.